app/src/main/java/org/simple/clinic/widgets/BindingPagingItemAdapter.kt

```diff
@@ -0,0 +1,50 @@
+package org.simple.clinic.widgets
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.annotation.LayoutRes
+import androidx.paging.PagedListAdapter
+import androidx.recyclerview.widget.DiffUtil
+import androidx.viewbinding.ViewBinding
+import io.reactivex.Observable
+import io.reactivex.subjects.PublishSubject
+import io.reactivex.subjects.Subject
+import org.simple.clinic.widgets.recyclerview.BindingViewHolder
+
+// TODO (SM) : Rename to PagingItemAdapter once migration to ViewBinding is finished
+open class BindingPagingItemAdapter<I : BindingPagingItemAdapter.Item<E>, E>(
+    diffCallback: DiffUtil.ItemCallback<I>,
+    private val bindings: Map<Int, (layoutInflater: LayoutInflater, parent: ViewGroup) -> ViewBinding>
+) : PagedListAdapter<I, BindingViewHolder>(diffCallback) {
+
+  private val eventSubject = PublishSubject.create<E>()
+  val itemEvents: Observable<E> = eventSubject.hide()
+
+  override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BindingViewHolder {
+    val layoutInflater = LayoutInflater.from(parent.context)
+
+    val binding = bindings
+        .filter { it.key == viewType }
+        .values
+        .first()
```

Comment:
Since `bindings` is a map, can't we just use `bindings.getValue(viewType)` here?

Reply:
Yep, I forgot about that extension 😅